src/vis/bar/utils.ts

```diff
@@ -0,0 +1,240 @@
+
+            const uniqueGroupVals = [...new Set(currColumn.vals.map((v) => v.val))];
+            const uniqueColVals = [...new Set(catCurr.vals.map((v) => v.val))];
+
+            uniqueGroupVals.forEach((uniqueVal) => {
+
+                const groupedLength = uniqueColVals.map((v) => {
+                    const allObjs = catCurr.vals.filter((c) => c.val === v).map((c) => c.id);
+                    const allGroupObjs = currColumn.vals.filter((c) => c.val === uniqueVal).map((c) => c.id);
+                    const joinedObjs = allObjs.filter((c) => allGroupObjs.includes(c));
+
+                    return normalizedFlag ? joinedObjs.length / allObjs.length * 100 : joinedObjs.length;
+                });
+
+                plots.push({
+                    data: {
+                        x: vertFlag ? [...new Set(catCurr.vals.map((v) => v.val))] : groupedLength,
+                        y: !vertFlag ? [...new Set(catCurr.vals.map((v) => v.val))] : groupedLength,
+                        orientation: vertFlag ? 'v' : 'h',
+                        xaxis: counter === 1 ? 'x' : 'x' + counter,
+                        yaxis: counter === 1 ? 'y' : 'y' + counter,
+                        showlegend: counter === 1 ? true : false,
+                        type: 'bar',
+                        name: uniqueVal,
+                    },
+                    xLabel: vertFlag ? catCurr.info.name : normalizedFlag ? 'Percent of Total' : 'Count',
+                    yLabel: vertFlag ? normalizedFlag ? 'Percent of Total' : 'Count' : catCurr.info.name
+                });
+                counter += 1;
+            });
+
+        } else {
+            const count = [...new Set(catCurr.vals.map((v) => v.val))].map((curr) => catCurr.vals.filter((c) => c.val === curr).length);
+            const valArr = [...new Set(catCurr.vals.map((v) => v.val))];
+            plots.push({
+                data: {
+                    x: vertFlag ? valArr : count,
+                    y: !vertFlag ? valArr : count,
+                    orientation: vertFlag ? 'v' : 'h',
+                    xaxis: counter === 1 ? 'x' : 'x' + counter,
+                    yaxis: counter === 1 ? 'y' : 'y' + counter,
+                    type: 'bar',
+                    name: catCurr.info.name
+                },
+                xLabel: vertFlag ? catCurr.info.name : normalizedFlag ? 'Percent of Total' : 'Count',
+                yLabel: vertFlag ? normalizedFlag ? 'Percent of Total' : 'Count' : catCurr.info.name
+            });
+            counter += 1;
+        }
+    }
+
+    const rows = Math.ceil(Math.sqrt(counter - 1));
+    const cols = Math.ceil((counter - 1) / rows);
+
+    return {
+        plots,
+        legendPlots: [],
+        rows,
+        cols,
+        errorMessage: 'To create a Bar Chart, please select at least 1 categorical column.',
+        formList: ['groupBy', 'barMultiplesBy', 'barDirection', 'barGroupType', 'barNormalized']
```

Comment:
```suggestion
```

Can we remove this properties if we throw errors (as suggested above)?

Reply:
even without the errors this was not necessary any more, removed it